samza-core/src/main/java/org/apache/samza/zk/ZkLeaderElector.java

```diff
@@ -84,6 +84,7 @@ private String getHostName() {
 
   @Override
   public void tryBecomeLeader(LeaderElectorListener leaderElectorListener) {
+    zkUtils.makeSurePersistentPathsExists(new String[]{keyBuilder.getProcessorsPath()});
```

Comment:
@sborya moved the commented-out logic to ZkControllerImpl:54. I guess he wanted to create all the zk nodes needed for ZkCoordinator together there. However, this doesn't work for coordinator service which has ZkLeaderElector but not ZkController. So I added this logic back here. Maybe we can clean up the creation of these paths by:
1) ZkLeaderElector ensures the processor paths exist
2) ZkBarrierForVersionUpdate ensures the job model version paths exist

This way they can be used independently. What do you think?